packages/components/src/combobox-control/index.tsx

```diff
@@ -258,9 +266,13 @@ function ComboboxControl( {
 								onFocus={ onFocus }
 								onBlur={ onBlur }
 								isExpanded={ isExpanded }
-								selectedSuggestionIndex={ matchingSuggestions.indexOf(
+								selectedSuggestionIndex={
 									selectedSuggestion
-								) }
+										? matchingSuggestions.indexOf(
+												selectedSuggestion
+										  )
+										: 0
```

Comment:
The `0` is arbitrary, and maybe wrong.

But there's a TS error without it:

```
Argument of type 'Option | null' is not assignable to parameter of type 'Option'.
  Type 'null' is not assignable to type 'Option'.ts(2345)
```

Reply:
If `selectedSuggestion` is null, you can't look for it in `matchingSuggestions`:

```jsx
matchingSuggestions.indexOf( selectedSuggestion )
```

Reply:
The type of `matchingSuggestions` is `Option[]`.